src/seamless-immutable.js

```diff
@@ -471,8 +471,17 @@
     return makeImmutable(obj, mutatingObjectMethods);
   }
 
+  var REACT_ELEMENT_TYPE = typeof Symbol === 'function' && Symbol.for && Symbol.for('react.element');
+  var REACT_ELEMENT_TYPE_FALLBACK = 0xeac7;
+
+  function isReactElement(obj) {
+    return obj.hasOwnProperty &&
+           obj.hasOwnProperty('$$typeof') &&
+           obj.$$typeof === REACT_ELEMENT_TYPE_FALLBACK || obj.$$typeof === REACT_ELEMENT_TYPE;
+  }
```

Comment:
Hopefully this isn't something that will change in React, since it was
added as a security feature. I'm not sure it's possible to depends on
multiple versions of the same package with npm, so might need to use
something else (bower?) to work around that.

I'll add some more comments and some tests for react 0.14 as a first pass.

I agree that it's strange for an immutability library to special case
react, but it does make working with the library more, well, seamless :)

On Thu, 14 Apr 2016 at 14:45 Richard Feldman <notifications@github.com>
wrote:

> In src/seamless-immutable.js
> <https://github.com/rtfeldman/seamless-immutable/pull/120#discussion_r59730057>
> :
>
> > @@ -471,8 +471,17 @@
> >      return makeImmutable(obj, mutatingObjectMethods);
> >    }
> >
> > +  var REACT_ELEMENT_TYPE = typeof Symbol === 'function' && Symbol.for && Symbol.for('react.element');
> > +  var REACT_ELEMENT_TYPE_FALLBACK = 0xeac7;
> > +
> > +  function isReactElement(obj) {
> > +    return obj.hasOwnProperty &&
> > +           obj.hasOwnProperty('$$typeof') &&
> > +           obj.$$typeof === REACT_ELEMENT_TYPE_FALLBACK || obj.$$typeof === REACT_ELEMENT_TYPE;
> > +  }
>
> This is looking like the right approach! Would you mind adding some
> comments to the detection linking to where the magic constants come from?
>
> I tried to avoid special casing React components for a long time, but the
> overwhelming demand has convinced me otherwise. One of the things this
> means is that now we need to document which React version(s) are supported,
> and the test cases will need to back that up.
>
> So it probably won't be as simple as just adding React as a single dev
> dependency; since React will keep having new releases, there will pretty
> quickly need to be multiple React versions supported and tested separately!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/rtfeldman/seamless-immutable/pull/120/files/e5602f933b39e26c2f8ee78aaddd221745be27c8#r59730057>
>